### ISSUE TITLE
Add optional flag so that sending doesn't always create two outputs

### ISSUE THREAD

**Issue by pablolarrimbe** [bitpay/copay#6658]
Hello Team,

I'd like to be able to avoid sending two outputs as an optional flag.  

Current behaviour:

Let's say I have 1 BTC in address 123.  I want to send 0.5 BTC to address 124.  Copay will send 0.5 to 124 and 0.5 to 125, with the latter being an address under that same wallet.  One input, two outputs.

Wanted behaviour:

When the "do not send all balance" (or something similar) is on, Copay will just create a transaction to 124 with 0.5 and 123 will keep the remaining 123.  One input, one output.

Reasons:

First, I'd like to control the behaviour of how things work.
Second, in times where transaction fees are extremely expensive, this will at least help me reduce the size of the transaction and with that the amount of fees to be payed.  Also, thinking for the network as a whole, it will save space and allow more transaction throughput into the blockchain.
Third, some people like to just keep everything under one address and not have to bother with having their BTC spread across multiple addresses.

To close, I know that creating a new address every time is a feature and I'm by no means asking for this to go away, just to have the option to opt out from this behaviour.

Thanks,
Pablo.

**Comment by dabura667** [comment]
This is not possible in Bitcoin.

**Comment by pablolarrimbe** [comment]
OK, but can we at least have the option to return the remaining to the same address?

**Comment by dabura667** [comment]
It already exists.

**Comment by pablolarrimbe** [comment]
Can you please let me know how to do it? I can't find it.

It'd be good to have that option on "receiving" too.

**Comment by JDonadio** [comment]
Copay does not "understand" about main and change addresses, just say "create this transaction" and `bws` does the rest.

https://github.com/bitpay/bitcore-wallet-service/blob/master/lib/server.js#L1988

**Comment by pablolarrimbe** [comment]
I don't understand.  Based on my initial example are you saying there is no way to implement a feature so that I can tell copay:

From my 1.00 BTC, send 0.5 to this address and the remaining to this other address of my choice?  In that case that other address of my choice can be an automatically generated address from Copay or I can override that and point it back to my own existing address? 

The same for the feature "receiving".  Copay knows my addresses.  Why can't I have a selector which by default shows a new address and if I open it shows the existing addresses with the balance so I can choose to receive BTC to an existing address.  Copay already knows which are my addresses with balance.

**Comment by JDonadio** [comment]
I mean that kind of change should be done in `bitcore-wallet-service`, not here.

**Comment by pablolarrimbe** [comment]
Isn't everything under the same umbrella?  Anyways, should I file the bug in that repo?  What's your recommendation?

**Comment by matiu** [comment]
> > Hello Team,
> 
> I'd like to be able to avoid sending two outputs as an optional flag.
> 
> Current behaviour:
> 
> Let's say I have 1 BTC in address 123. I want to send 0.5 BTC to address 124. Copay will send 0.5 to 124 and 0.5 to 125, with the latter being an address under that same wallet. One input, two outputs.
> 
> Wanted behaviour:
> 
> When the "do not send all balance" (or something similar) is on, Copay will just create a transaction to 124 with 0.5 and 123 will keep the remaining 123. One input, one output.


Hola Pablo!

That is no possible. In Bitcoin, each UTXO (unspent output) must be used in full in a transaction. That is why a "change" address have to be specified in the transaction.

Copay offers "single address wallets" (set a wallet creation time) in which a single address will be over an over. That is usually a bad idea (for privacy reasons) but it is OK for certain special use cases (organizations what want their wallets to be auditable)

**Comment by pablolarrimbe** [comment]
Hola Mat :)

Yes I got that bit.  That was my bad.  But I think the discussion after that still stands.  Can't Copay let me choose what that "change" address will be?  It sounds like it shouldn't be a big deal.  Copay is already choosing an address within my seed so the change should be to just allow the user to optionally change that default and if the user chooses to, they can pick from a selector which address would like to use.

And the same thing for receiving.  I would like to be able to choose where I want to receive it.

I also agree that one single address is a bad idea, but also when dealing with multiple small payments it starts getting a bit confusing with so many addresses, etc.

Just my $0.02 here of what I've observed as a user and I think this change would just add more flexibility without really hurting usability.
